open-apparel-registry/open-apparel-registry
# Fix claim showing incorrect parent facility

## Overview

The claim facility form was correctly submitting and saving the
contributor name and contributor id; however, when serializing the data
for fetching it, we were incorrectly grabbing the user id. This caused
the form to display the incorrect contributor in the dropdown.

Claims now correctly serialize the parent_facility with the contributor
id, not the admin user id.

We have run into similar bugs many times, because in development, the 
default contributor and user IDs match. This has
caused us to miss a number of bugs related to mixing up contributors
and users when working with data (because these two entities are used
somewhat interchangeably). In addition to updating the claim serialization,
the IDs of contributors and their associated users have been updated in our 'fixtures' files,
and now will no longer match after running `resetdb`, which will
hopefully prevent this category of bug.

Connects #1685 

## Demo

<img width="430" alt="Screen Shot 2022-03-02 at 11 24 56 AM" src="https://user-images.githubusercontent.com/21046714/156405890-5821c99e-9bb1-4ff4-82c2-f2cc54dee25c.png">

## Testing Instructions

* Checkout this branch, then checkout commit 1073cd3. Run `./scripts/resetdb`. 
* Navigate to a claim and save a parent company using the form, then refresh the page. Note that the displayed parent company is not the one you saved. 
* View the same claim in the Django admin and confirm that the stored parent company is correct.
* Checkout commit 4ae4282 view the same claim's claim form. Confirm that the displayed parent company is correct. 
* Resave with a new parent company and refresh the page and confirm that the correct parent company is saved and displayed. 

## Checklist
